contracts/contracts/RequestManager.sol

```diff
@@ -66,53 +83,99 @@ contract RequestManager is Ownable {
         bytes32 fillId
     );
 
+    /// Emitted when staked native tokens tied to a claim have been withdrawn.
+    ///
+    /// This can only happen when the claim has been resolved and the caller
+    /// of :sol:func:`withdraw` is allowed to withdraw their stake.
     event ClaimWithdrawn(
         uint256 claimId,
         uint256 indexed requestId,
         address claimReceiver
     );
 
     // Constants
+
+    /// The minimum amount of source chain's native token that the claimer needs to
+    /// provide when making a claim, as well in each round of the challenge game.
     uint256 public claimStake;
+
+    /// The period for which the claim is valid.
     uint256 public claimPeriod;
+
+    /// The period by which the termination time of a claim is extended after each
+    /// round of the challenge game. This period should allow enough time for the
+    /// other parties to counter-challenge.
+    ///
+    /// Note that in the first challenge round, i.e. the round initiated by the first
+    /// challenger, the termination time is extended additionally by the finalization
+    /// time of the target chain. This is done to allow for L1 resolution.
     uint256 public challengePeriodExtension;
 
+    /// The minimum validity period of a request.
     uint256 public constant MIN_VALIDITY_PERIOD = 5 minutes;
+
+    /// The maximum validity period of a request.
     uint256 public constant MAX_VALIDITY_PERIOD = 52 weeks;
 
     // Variables
+
+    /// Indicates whether the contract is deprecated. A deprecated contract
+    /// cannot be used to create new requests.
     bool public deprecated;
+
+    /// The request counter, used to generate request IDs.
     uint256 public requestCounter;
+
+    /// The claim counter, used to generate claim IDs.
     uint256 public claimCounter;
+
+    /// The resolution registry that is used to query for results of L1 resolution.
     ResolutionRegistry public resolutionRegistry;
 
-    mapping(uint256 => uint256) public finalizationTimes; // target rollup chain id => finalization time
+    /// Maps target rollup chain IDs to finalization times.
+    /// Finalization times are in seconds.
+    mapping(uint256 => uint256) public finalizationTimes;
 
+    /// Maps request IDs to requests.
     mapping(uint256 => Request) public requests;
+
+    /// Maps claim IDs to claims.
     mapping(uint256 => Claim) public claims;
 
+    /// The minimum fee paid to the liquidity provider.
```

Comment:
To prevent confusion, would also be good to state that this value is denominated in transfer token.